packages/builder/src/pages/builder/_layout.svelte

```diff
@@ -1,36 +1,162 @@
+  import { writable } from "svelte/store"
 
-  let loaded = false
+  let initPromise
+  let loaded = writable(false)
   let commandPaletteModal
   let settingsModal
+  let accountLockedModal
+  let hasAuthenticated = false
 
   $: multiTenancyEnabled = $admin.multiTenancy
   $: hasAdminUser = $admin?.checklist?.adminUser?.checked
   $: baseUrl = $admin?.baseUrl
-  $: tenantSet = $auth.tenantSet
   $: cloud = $admin?.cloud
   $: user = $auth.user
-
+  $: isOwner = $auth.accountPortalAccess && $admin.cloud
   $: useAccountPortal = cloud && !$admin.disableAccountPortal
+  $: isBuilder = sdk.users.hasBuilderPermissions(user)
+
+  $: usersLimitLockAction = $licensing?.errUserLimit
+    ? () => accountLockedModal.show()
+    : null
 
   navigation.init($redirect)
 
+  // Only recalculates when an action actually changes
+  const navigationAction = derivedMemo(
+    [admin, auth, enrichedApps, appStore, isActive, appsStore, loaded],
+    ([
+      $admin,
+      $auth,
+      $enrichedApps,
+      $appStore,
+      $isActive,
+      $appsStore,
+      $loaded,
+    ]) => {
+      // Only run logic when fully loaded
+      if (!$loaded || !$admin.loaded || !$auth.loaded) {
+        return null
+      }
+
+      // Return URL management - set when *not* logged in
+      // will redirect back to this location on auth
+      if (
+        !$auth.user &&
+        !CookieUtils.getCookie(Constants.Cookies.ReturnUrl) &&
+        !$auth.postLogout &&
+        !$isActive("./auth") &&
+        !$isActive("./invite") &&
+        !$isActive("./admin")
+      ) {
+        return { type: "setReturnUrl", url: window.location.pathname }
+      }
+
+      // if tenant is not set go to it
+      if (!useAccountPortal && multiTenancyEnabled && !$auth.tenantSet) {
+        return { type: "redirect", path: "./auth/org" }
+      }
+
+      // Force creation of an admin user if one doesn't exist
+      if (!useAccountPortal && !hasAdminUser) {
+        return { type: "redirect", path: "./admin" }
+      }
+
+      // Redirect to log in at any time if the user isn't authenticated
+      if (
+        (hasAdminUser || cloud) &&
+        !$auth.user &&
+        !$isActive("./auth") &&
+        !$isActive("./invite") &&
+        !$isActive("./admin")
+      ) {
+        return { type: "redirect", path: "./auth" }
+      }
+
+      // Check if password reset required for user
+      if ($auth.user?.forceResetPassword) {
+        return { type: "redirect", path: "./auth/reset" }
+      }
+
+      // Authenticated user navigation
+      if ($auth.user) {
+        const returnUrl = CookieUtils.getCookie(Constants.Cookies.ReturnUrl)
+
+        // Return to saved URL first - skip onboarding check if user has a return URL
+        if (returnUrl) {
+          return { type: "returnUrl", url: returnUrl }
+        }
+
+        // Builders without apps should be redirected to the onboarding flow.
```

Comment:
As discussed, we will want to change this approach. Onboarding should be reserved for new accounts. This scenario is a "no workspaces" case, which has a slightly different context. 
I.e. it's not necessarily the "first" workspace, etc.